django_tenants/models.py

```diff
@@ -127,16 +126,11 @@ class DomainMixin(models.Model):
     def save(self, *args, **kwargs):
         # Get all other primary domains with the same tenant
         domain_list = self.__class__.objects.filter(tenant=self.tenant, is_primary=True).exclude(pk=self.pk)
-        if len(domain_list) == 0:
-            # We have no primary domain yet, so set as primary domain by default
-            self.is_primary = True
-        else:
-            if self.is_primary:
-                # Remove primary status of existing domains for tenant
-                for domain in domain_list:
-                    domain.is_primary = False
-                    domain.save()
-
+        # If we have no primary domain yet, set as primary domain by default
+        self.is_primary = self.is_primary or (not domain_list.exists())
+        if self.is_primary:
+            # Remove primary status of existing domains for tenant
+            domain_list.update(is_primary=False)
```

Comment:
If you're trying the to save a domain that's already has `is_primary=True` this command will set it to `False` and then the super in the next line will set it to `True` again, right? Or should we do something like 
`domain_list.exclude(id=self.id).update(is_primary=False)`